### PR TITLE
fix presentation

### PR DESCRIPTION
**Pull request by berendsliedrecht**
- Draft as we only support openidtokenissuer as DID, and we want to use JWK here.

### CODE REVIEW

File: agent/src/endpoints.ts [hunk 1]
```diff
@@ -217,7 +228,10 @@ apiRouter.post('/requests/receive', async (request: Request, response: Response)
     resolved.presentationExchange?.credentialsForRequest
   )
 
+  const jwk = getJwkFromKey(await agent.context.wallet.createKey({ keyType: KeyType.P256 }))
```
Comment (berendsliedrecht) [inline agent/src/endpoints.ts:231]: currently this is broken :/

Comment (berendsliedrecht) [reply agent/src/endpoints.ts]: NOT ANYMORE

File: app/components/ReceiveTab.tsx [hunk 2]
```diff
@@ -3,24 +3,28 @@ import { Button } from '@/components/ui/button'
...
   const [receivedCredentials, setReceivedCredentials] = useState()
+  const [receivedPresentation, setReceivedPresentation] = useState()
 
   async function onSubmitReceiveOffer(e: FormEvent) {
     e.preventDefault()
-    if (!receiveCredentialOfferUri) return
-
-    setReceivedCredentials(await receiveOffer(receiveCredentialOfferUri))
+    if (!receiveCredentialOfferOrPresentationUri) return
+    if (receiveCredentialOfferOrPresentationUri.startsWith('openid-credential-offer://')) {
```
Comment (TimoGlastra) [inline app/components/ReceiveTab.tsx:16]: I would not check based on the scheme, as we for paradym use https://paradym.id so we can use deep linking. And e.g. HAIP uses `As a way to invoke the Wallet, at least a custom URL scheme haip:// MUST be supported`.

So i think we should check whether `credential_offer` or `credential_offer_uri` params are present and extract those. That is the most robust method I think

**Review by TimoGlastra — APPROVED** [review]

File: app/components/ReceiveTab.tsx (hunk 2 above)
Comment (TimoGlastra) [reply app/components/ReceiveTab.tsx:16]: Can't open an issue for this as we're in a fork

**Pull request merged**